services/QuillLMS/spec/controllers/concerns/new_relic_attributable_spec.rb

```diff
@@ -34,4 +34,19 @@ def index
       get :index
     end
   end
+
+  context 'Evidence engine' do 
+    controller(Evidence::ApiController) do
+  
+      def index
+        head :ok
+      end
+    end
+
+    it 'should pass user_id to NewRelic' do
+      allow(controller).to receive(:current_user) { user }
```

Comment:
Have you tested this on a staging environment or with New Relic locally? I think this test setup (stubbing `current_user` and putting the test in the LMS) might be covering up that `current_user` isn't accessible by `Evidence::ApiController` controllers and that the `New::Relic` agent also may not be accessible.

It might we'd need to pass the New Relic agent to the engine [here](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/config/initializers/evidence.rb) and copy and re-write the concern for the engine (it's duplication, but it's only a few lines of code). You should be able to put a test in the Evidence engine that uses the dummy app and passes. That should be the threshold.

Reply:
> Have you tested this on a staging environment or with New Relic locally? I think this test setup (stubbing `current_user` and putting the test in the LMS) might be covering up that `current_user` isn't accessible by `Evidence::ApiController` controllers and that the `New::Relic` agent also may not be accessible.
> 
> It might we'd need to pass the New Relic agent to the engine [here](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/config/initializers/evidence.rb) and copy and re-write the concern for the engine (it's duplication, but it's only a few lines of code). You should be able to put a test in the Evidence engine that uses the dummy app and passes. That should be the threshold.

Thanks for raising these concerns. Here is a demonstration of this PR currently working as is:
<img width="524" alt="Screen Shot 2021-12-15 at 1 51 31 PM" src="https://user-images.githubusercontent.com/90669/146263258-c0976c63-d564-459c-a5b5-98975d0d9ddd.png">

Reply:
link from the screenshot: https://one.newrelic.com/launcher/nr1-core.explorer?platform[timeRange][duration]=1800000&pane=eyJuZXJkbGV0SWQiOiJlcnJvcnMtdWkub3ZlcnZpZXciLCJiYXJjaGFydCI6ImJhcmNoYXJ0IiwidG9wRmFjZXQiOiJ0cmFuc2FjdGlvblVpTmFtZSIsInBhZ2UiOiJzaG93IiwicHJpbWFyeUZhY2V0IjoiZXJyb3IuY2xhc3MiLCJ0cmFjZUlkIjoiODIyNGNjMTYtNWRlOC0xMWVjLWI5YzEtMDI0MmFjMTEwMDBhXzBfNjg1NyIsImVudGl0eUd1aWQiOiJNall6T1RFeE0zeEJVRTE4UVZCUVRFbERRVlJKVDA1OE5UVXhPRFE0TVRRdyJ9&sidebars[0]=eyJuZXJkbGV0SWQiOiJucjEtY29yZS5hY3Rpb25zIiwiZW50aXR5R3VpZCI6Ik1qWXpPVEV4TTN4QlVFMThRVkJRVEVsRFFWUkpUMDU4TlRVeE9EUTRNVFF3Iiwic2VsZWN0ZWROZXJkbGV0Ijp7Im5lcmRsZXRJZCI6ImVycm9ycy11aS5vdmVydmlldyJ9fQ==&state=6be8de5e-a806-144b-0a00-cb44bca49115